app/fireweather_bot.py

```diff
@@ -0,0 +1,276 @@
+    index_names = data_df[data_df['status'] == 'actual'].index
+    data_df.drop(index_names, inplace=True)
+    # can now delete the 'status' column entirely, since we know it's all forecasts
+    # and we don't want to write this column to the DB
+    data_df.drop(['status'], axis=1, inplace=True)
+    # weather_date is formatted yyyymmdd as an int - need to reformat it as a Date
+    data_df['weather_date'] = pd.to_datetime(
+        data_df['weather_date'], format='%Y%m%d')
+    # delete the temp CSV file - it's not needed anymore
+    os.remove(os.path.join(temp_path, TEMP_CSV_FILENAME))
+    # write to database using _session's engine
+    session = app.db.database.get_session()
+    # write the data_df to the database one row at a time, so that if data_df contains >=1 rows that are
+    # duplicates of what is already in the db, the write won't fail for the unique rows
+    # NOTE: iterating over the data_df one Series (row) at a time is ugly, but until pandas is updated
+    # with a fix, this is the easiest work-around.
+    # See https://github.com/pandas-dev/pandas/issues/15988
+    for index, row in data_df.iterrows():
+        try:
+            row = row.dropna()
+            session.add(NoonForecasts(**row.to_dict()))
+            session.flush()
+            session.commit()
+        except IntegrityError:
+            LOGGER.info('Skipping duplicate record')
+            session.rollback()
+
+def _get_now():
+    """ Helper function to get the current time (easy function to mock out in testing) """
+    return datetime.now(tz=pytz.utc)
+
+
+def _get_start_date():
+    """ Helper function to get the start date for query (if morning run, use current day; if evening run,
+    use tomorrow's date, since we only want forecasts, not actuals)
+    Strip out time, we just want yyyymmdd """
+    date = ''
+    now = _get_now()
+    pacific_time = datetime.now(tz=pytz.timezone('Canada/Pacific'))
+    if pacific_time.hour < 11:
+        date = now
+    else:
+        date = now + timedelta(days=1)  # use tomorrow's date
+    return date.strftime('%Y%m%d')
+
+
+def _get_end_date():
+    """ Helper function to get the end date for query (5 days in future).
+    Strip out time, we just want <year><month><date> """
+    five_days_ahead = _get_now() + timedelta(days=5)
+    return five_days_ahead.strftime('%Y%m%d')
+
+
+def _get_stations_local() -> List[WeatherStation]:
+    """ Get list of stations from local json files.
+    """
+    with open(weather_stations_file_path) as weather_stations_file:
+        json_data = json.load(weather_stations_file)
+        return json_data['weather_stations']
+
+
+def _get_station_names_to_codes() -> Dict:
+    """ Helper function to create dictionary of (station_name: station_code) key-value pairs
+    Is used when replacing station names with station IDs in dataframe
+    """
+    station_data = _get_stations_local()
+    station_codes = {}
```

Comment:
Sure - that works - but how sexy is this: `station_codes = { station['name'] : station['code'] for station in station_data}`